store/precomputed_key/wvm/rpc/wvm_rpc_client.go

```diff
@@ -0,0 +1,241 @@
+package wvm
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	wvmtypes "github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/wvm/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type Signer interface {
+	GetAccount(ctx context.Context) (common.Address, error)
+	SignTransaction(ctx context.Context, signData *wvmtypes.SignData) (string, error)
+}
+
+// WVM RPC client
+type RPCClient struct {
+	log     log.Logger
+	client  *ethclient.Client
+	chainID int64
+	signer  Signer
+}
+
+func NewWvmRPCClient(log log.Logger, cfg *wvmtypes.Config, signer Signer) (*RPCClient, error) {
+	client, err := ethclient.Dial(cfg.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to the WVM client: %w", err)
+	}
+
+	ethRPCClient := &RPCClient{
+		log:     log,
+		client:  client,
+		chainID: cfg.ChainID,
+		signer:  signer,
+	}
+
+	return ethRPCClient, nil
+}
+
+func (rpc *RPCClient) SendTransaction(ctx context.Context, to string, data []byte) (string, error) {
+	gas, err := rpc.estimateGas(ctx, to, data)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed estimate gas: %w", err)
+	}
+
+	wvmRawTx, err := rpc.createRawTransaction(ctx, to, string(data), gas)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed create transaction: %w", err)
+	}
+
+	wvmTxHash, err := rpc.sendRawTransaction(ctx, wvmRawTx)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed to send transaction: %w", err)
+	}
+
+	return wvmTxHash, nil
+}
+
+// estimateGas tries estimates the suggested amount of gas that required to execute a given transaction.
+func (rpc *RPCClient) estimateGas(ctx context.Context, to string, data []byte) (uint64, error) {
+	var (
+		toAddr    = common.HexToAddress(to)
+		bytesData []byte
+		err       error
+	)
+
+	fromAddress, err := rpc.signer.GetAccount(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to estimate gas, no signer")
+	}
+
+	var encoded string
+	if string(data) != "" {
+		if ok := strings.HasPrefix(string(data), "0x"); !ok {
+			encoded = hexutil.Encode(data)
+		}
+
+		bytesData, err = hexutil.Decode(encoded)
+		if err != nil {
+			return 0, err
+		}
+	}
+
+	msg := ethereum.CallMsg{
+		From: fromAddress,
+		To:   &toAddr,
+		Gas:  0x00,
```

Comment:
why provide 0 gas?

Reply:
mmm, in this case just makes things explicit
as CallMsg says:   
```
Gas       uint64          // if 0, the call executes with near-infinite gas
```